### DISCUSSION TITLE
[Swift 5.10] Build failure when swift-case-paths version resolves to anything beyond 1.0.0 - error: macros are an experimental feature that is not enabled

### DISCUSSION THREAD

**Discussion by clackary** [pointfreeco/swift-parsing#331] [Q&A]
After digging a bit, this is mostly a heads up that a [new compiler error](https://github.com/apple/swift/commit/becbdc8203b0ff7410f88a1ff61d3f041349a623) in recent toolchains, possibly as early as 5.9.1, seems to break `swift-case-paths` build.

```
./swift-parsing/.build/checkouts/swift-case-paths/Sources/CasePaths/Macros.swift:29:13: error:  macros are an experimental feature that is not enabled 
  @attached(extension, conformances: CasePathable)
            ^
```

To reproduce:
 - Download and install a dev toolchain (I ran with `swift-DEVELOPMENT-SNAPSHOT-2023-07-11-a`, ~but I'm guessing `swift-5.9.1-RELEASE` would likely also reproduce~)
 - Pin `swift-case-paths` to something from 1.1.0 - `.package(url: "https://github.com/pointfreeco/swift-case-paths", from: "1.1.0"),`
 - `swift build`
 
 Workaround:
 - Pin `swift-case-paths` to 1.0.0 - `.package(url: "https://github.com/pointfreeco/swift-case-paths", exact: "1.0.0"),`
 
 
 Edit: Updated commit and dev toolchain which reproduces the error.

**Comment by clackary** [comment]
Let me know if this issue belongs over in swift-case-paths, but since the workaround can be applied in swift-parsing (or also in whatever package includes swift-parsing), I figured it makes sense to file here.

**Comment by stephencelis** [comment]
@clackary We run CI against Swift 5.9.1, so I don't think this is an issue with the library so much as an issue with development snapshots needing to explicitly enable certain feature flags. If you can reproduce this with an official release and share the repro, please do! Till we can determine this is a bug in the library I'm going to convert it to a more general discussion.

**Reply by clackary** [reply]
Good call, just checked swift-5.9.1-RELEASE toolchain, and did not reproduce with that.

**Comment by clackary** [comment]
Quick update here -- I took another swing at this, and I've narrowed it down a bit.

Building swift-parsing library with 2023-07-11 toolchain and swift-case-paths:1.1.3 exhibits the 'experimental feature' error.
Building swift-parsing library with 2023-10-02 toolchain and swift-case-paths:1.1.3 builds successfully.

I checked the latest available toolchain (2023-12-07) as well, and it built the library successfully. 

I will note that I ran into a different error when attempting to build the entire package, but that is likely a discussion for another time. Looks like a compiler bug when building a benchmark...
```
./swift-parsing/Sources/swift-parsing-benchmark/JSON.swift:71:84: error: failed to produce diagnostic for expression; please submit a bug report (https://swift.org/contributing/#reporting-bugs)
      var body: some ParserPrinter<Substring.UTF8View, [String: JSONValue.Output]> {
                                                                                   ^
```

For now it seems the experimental feature error isn't imminent though, so feel free to close.

**Reply by clackary** [reply]
Filed a compiler bug for the failed diagnostic: https://github.com/apple/swift/issues/70526
